### PR TITLE
fix(cosh): pass session ID to subprocesses

### LINKED ISSUE #1489: [TestAgent][cosh] bug: COSH_SESSION_ID is read from env but never exported to subprocess, breaking per-run correlation
## Summary

`copilot-shell` (cosh) 2.7.0 reads `COSH_SESSION_ID` from the parent process env (for inheritance / `--continue` / `--resume` precedence), but **never writes the resolved session id back to `process.env`**. As a result, child processes spawned by cosh — the shell tool, MCP servers, user `!cmd` escapes — do **not** see `COSH_SESSION_ID` in their environment.

This defeats the per-run cross-plane correlation that the surrounding comment in `packages/cli/src/config/config.ts` claims to provide:

> *"Adopt a launcher-provided correlation id (exported as COSH_SESSION_ID by the cosh wrapper) so this run's own session id matches what AgentSight scrapes from /proc/<pid>/environ, enabling per-run cross-plane correlation."*

AgentSight / external observers that scrape subprocess environ cannot associate a tool invocation back to its cosh run, because the env var is only ever *consumed*, never *produced*.

## Component

copilot-shell (cosh) 2.7.0

## Reproduction

1. Install copilot-shell 2.7.0 RPM.
2. Launch the interactive TUI:
   ```bash
   cosh
   ```
3. From inside the TUI, drop into a shell escape and ask for the env var:
   ```
   !printenv COSH_SESSION_ID
   ```
4. Observe the result box.

### Expected

The result box contains a non-empty UUID (the same `sessionId` cosh uses internally for chat-file naming, telemetry, and the debug log).

### Actual

The result box is **empty** — `COSH_SESSION_ID` is not present in the subprocess environment. Only `COSH_DATA_DIR` (set by the `/usr/bin/cosh` wrapper shell script) is visible:

```
!env | grep ^COSH
COSH_DATA_DIR=/usr/share/copilot-shell
```

## Root Cause

In `src/copilot-shell/packages/cli/src/config/config.ts`, `loadCliConfig` resolves the session id with the precedence:

```
--continue / --resume  >  COSH_SESSION_ID env  >  minted uuid (Config constructor)
```

…then passes it to `new Config({ sessionId, ... })`. But the resolved value is **never written back** to `process.env`. The only reference to `COSH_SESSION_ID` in the codebase is the read at `config.ts:965`:

```ts
if (!sessionId) {
  const envSessionId = process.env['COSH_SESSION_ID'];
  if (envSessionId && envSessionId.length > 0) {
    sessionId = envSessionId;
  }
}
```

There is no corresponding `process.env['COSH_SESSION_ID'] = sessionId` setter anywhere in `packages/cli` or `packages/core`. So:

- The cosh node process itself does not have `COSH_SESSION_ID` in its environ (env is captured at `exec` time, before the id is minted).
- Every child process cosh spawns inherits from `process.env`, which still lacks `COSH_SESSION_ID`.

## Suggested Fix

Write the resolved id back to `process.env` right after `Config` is constructed, so child processes inherit it. Idempotent — no-op when the id was inherited from the parent env.

```diff
--- a/src/copilot-shell/packages/cli/src/config/config.ts
+++ b/src/copilot-shell/packages/cli/src/config/config.ts
@@ -1109,6 +1109,15 @@ export async function loadCliConfig(
     }
   }
 
+  // Export the resolved session id so child processes (shell tool, MCP
+  // servers, user `!cmd` escapes) inherit COSH_SESSION_ID, enabling per-run
+  // cross-plane correlation with AgentSight. Idempotent: no-op when the id
+  // was already inherited from the parent env.
+  const resolvedSessionId = config.getSessionId();
+  if (resolvedSessionId) {
+    process.env['COSH_SESSION_ID'] = resolvedSessionId;
+  }
+
   return config;
 }
```

Note: this makes child subprocess environ carry `COSH_SESSION_ID`, which is what the test (and the code comment) actually want. Scraping `/proc/<cosh_node_pid>/environ` directly still won't show the id (environ is frozen at exec); a wrapper-level fix would be needed if that is also desired, but that is out of scope here.

## Verification

Applied the patch, rebuilt `copilot-shell` RPM, reinstalled, and re-ran the reproduction. After the fix, `!printenv COSH_SESSION_ID` returns a stable UUID that matches across repeated invocations within the same cosh session, and is the same id cosh uses for the chat jsonl file under `~/.copilot-shell/...`.

A regression test covering this (TC-3.1: *子进程 `!printenv COSH_SESSION_ID` 存在且稳定*) previously had to be marked `it.fails()`; with the patch applied it now passes.

## Related

- PR #1383 introduced the env-read half but did not land the export half.
- Issue #481 (CLOSED, feat) exposed `run_id` in `HookInput` for per-run correlation — related but orthogonal (hook input vs subprocess env).

### PR DESCRIPTION
**Pull request by kongche-jbw**
## Description

Propagate `COSH_SESSION_ID` explicitly to child processes created by shell execution and stdio MCP transports. The correlation ID is kept Config-scoped and pinned to the launcher ID instead of mutating global `process.env`, which prevents independently addressable ACP chats from reusing one internal session ID. When no launcher ID exists, the initial Config session ID is used as the fallback correlation ID.

## Related Issue

closes #1489

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] Documentation update
- [ ] Refactoring (no functional change)
- [ ] Performance improvement
- [ ] CI/CD or build changes

## Scope

- [x] `cosh` (copilot-shell)
- [ ] `cosh-ng` (cosh-ng)
- [ ] `sec-core` (agent-sec-core)
- [ ] `skill` (os-skills)
- [ ] `sight` (agentsight)
- [ ] `tokenless` (tokenless)
- [ ] `ckpt` (ws-ckpt)
- [ ] `memory` (agent-memory)
- [ ] `anolisa` (anolisa-cli)
- [ ] `skillfs` (SkillFS)
- [ ] Multiple / Project-wide

## Checklist

- [ ] I have read the [Contributing Guide](../CONTRIBUTING.md)
- [x] My code follows the project's code style
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have updated the documentation accordingly
- [x] For `cosh`: Lint passes, type check passes, and tests pass
- [x] Lock files are up to date (`package-lock.json` / `Cargo.lock`)

## Testing

- `npm run build`
- `npm run lint`
- `npm run typecheck`
- `npm test --workspace=packages/core -- --reporter=dot`
  - 3,783 passed, 2 skipped
- `npm test --workspace=packages/cli -- --reporter=dot`
  - Full CLI suite passed with no failures or errors
- Targeted session propagation tests
  - 424 passed, 2 skipped

Verified the following edge cases:

- A regular TUI session adopts the launcher-provided correlation ID.
- ACP chat sessions retain unique internal session IDs.
- Loading Config does not mutate global `process.env`.
- Shell tool, `!cmd`, prompt shell injection, PTY, and fallback execution receive `COSH_SESSION_ID`.
- Stdio MCP servers receive the Config-scoped environment.
- MCP server configuration cannot override the correlation ID.

## Additional Notes

This intentionally avoids the global `process.env` mutation proposed in #1490. A global mutation would cause subsequent ACP session configurations to re-adopt the same ID, breaking ACP session isolation.

No dependency or documentation files were changed.

### CODE REVIEW

**Review by samchu-zsl — APPROVED** [review]
本轮 assigned review 未发现新的 merge blocker。

本地验证覆盖了 Config-scoped `COSH_SESSION_ID`、shell execution、prompt shell、PTY/fallback execution 和 stdio MCP transport 的变更路径；targeted tests、core/cli build、core/cli typecheck 和 diff whitespace 检查均通过。
